rpc_util_test.go

```diff
@@ -290,3 +293,115 @@ func BenchmarkGZIPCompressor512KiB(b *testing.B) {
+	return nil, nil
+}
+
+func (m *MockCompressor) Decompress(r io.Reader) (io.Reader, error) {
+	if m.ErrDecompress != nil {
+		return nil, m.ErrDecompress
+	}
+	if m.CustomReader != nil {
+		return m.CustomReader, nil
+	}
+	return bytes.NewReader(m.DecompressedData), nil
+}
+
+func (m *MockCompressor) DecompressedSize(compressedBytes mem.BufferSlice) int {
+	return m.Size
+}
+
+func (m *MockCompressor) Name() string {
+	return "mockCompressor"
+}
+
+type ErrorReader struct{}
+
+func (e *ErrorReader) Read(p []byte) (n int, err error) {
+	return 0, errors.New("simulated io.Copy read error")
+}
+
+func TestDecompress(t *testing.T) {
+	tests := []struct {
+		name                  string
+		compressor            encoding.Compressor
+		input                 mem.BufferSlice
+		maxReceiveMessageSize int
+		wantedOutput          mem.BufferSlice
+		wantedSize            int
+		wantedError           error
+	}{
+		{
+			name: "Successful decompression",
+			compressor: &MockCompressor{
+				DecompressedData: []byte("decompressed data"),
+				Size:             17,
+			},
+			input:                 mem.BufferSlice{}, // You can fill in the actual input here
+			maxReceiveMessageSize: 100,
+			wantedOutput: func() mem.BufferSlice {
+				decompressed := []byte("decompressed data")
+				return mem.BufferSlice{mem.NewBuffer(&decompressed, nil)} // Taking the address of the slice variable
+			}(),
+			wantedSize:  1,
+			wantedError: nil,
+		},
+		{
+			name: "Error during decompression",
+			compressor: &MockCompressor{
+				ErrDecompress: errors.New("decompression error"),
+			},
+			input:                 mem.BufferSlice{},
+			maxReceiveMessageSize: 100,
+			wantedOutput:          nil,
+			wantedSize:            0,
+			wantedError:           errors.New("decompression error"),
+		},
+		{
+			name: "Buffer overflow",
+			compressor: &MockCompressor{
+				DecompressedData: []byte("overflow data"),
+				Size:             100,
+			},
+			input:                 mem.BufferSlice{},
+			maxReceiveMessageSize: 5, // Intentionally low to trigger overflow
+			wantedOutput:          nil,
+			wantedSize:            6,
+			wantedError:           errors.New("overflow: message larger than max size receivable by client (5 bytes)"),
+		},
+		{
+			name: "Error during io.Copy",
+			compressor: &MockCompressor{
+				CustomReader: &ErrorReader{},
+			},
+			input:                 mem.BufferSlice{}, // You can fill in the actual input here
+			maxReceiveMessageSize: 100,
+			wantedOutput:          nil,
+			wantedSize:            0,
+			wantedError:           errors.New("simulated io.Copy read error"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			output, size, err := decompress(tt.compressor, tt.input, tt.maxReceiveMessageSize, nil)
+
+			if tt.wantedError != nil {
+				require.EqualError(t, err, tt.wantedError.Error())
+			} else {
+				require.NoError(t, err)
+			}
+
+			require.Equal(t, tt.wantedSize, size)
```

Comment:
let's only use base equality `==` instead of `require.Equal`

Reply:
changed the required.Equal to ==